src/ExpressReceiver.ts

```diff
@@ -52,9 +72,49 @@ export default class ExpressReceiver implements Receiver {
 
     this.processBeforeResponse = processBeforeResponse;
     this.logger = logger;
-    const endpointList: string[] = typeof endpoints === 'string' ? [endpoints] : Object.values(endpoints);
+    const endpointList = typeof endpoints === 'string' ? [endpoints] : Object.values(endpoints);
+    this.router = Router();
     for (const endpoint of endpointList) {
-      this.app.post(endpoint, ...expressMiddleware);
+      this.router.post(endpoint, ...expressMiddleware);
+    }
+    this.app.use(this.router);
+    this.oauthAuthorize = undefined;
+
+    let oauthInstaller: OAuthInterface | undefined = undefined;
+    if (
+      clientId !== undefined
+      && clientSecret !== undefined
+      && (stateSecret !== undefined || stateStore !== undefined)
+    ) {
+
+      oauthInstaller = new InstallProvider({
+        clientId,
+        clientSecret,
+        stateSecret,
+        stateStore,
+        installationStore,
+        authVersion,
+      });
+    }
+
+    // Add OAuth routes to receiver
+    if (oauthInstaller !== undefined) {
+      this.app.use('/slack/install', (req, res) => {
+        return oauthInstaller!.handleCallback(req, res, {});
+      });
+      this.app.get('/slack/begin_auth', (_req, res, next) => {
+        // TODO, take in arguments or function for
+        oauthInstaller!.generateInstallUrl({
+          metadata,
+          scopes: scopes!,
+        }).then((url: string) => {
+          res.send(`<a href=${url}><img alt=""Add to Slack"" height="40" width="139"
+            src="https://platform.slack-edge.com/img/add_to_slack.png"
+            srcset="https://platform.slack-edge.com/img/add_to_slack.png 1x,
+            https://platform.slack-edge.com/img/add_to_slack@2x.png 2x" /></a>`);
+        }).catch(next);
```

Comment:
if you use `await`, wrap in `try`/`catch` instead. this seemed more reliable than `.catch(next)` in the work on global middleware.